src/d3d8/d3d8_device.h

```diff
@@ -723,7 +762,10 @@ namespace dxvk {
 
       D3D8VertexBuffer* buffer = static_cast<D3D8VertexBuffer*>(pStreamData);
 
-      m_streams[StreamNumber] = D3D8VBO {buffer, Stride};
+      const D3D8VBO& vbo = m_streams[StreamNumber];
+
+      if(likely(vbo.buffer.ptr() != buffer || vbo.stride != Stride))
+        m_streams[StreamNumber] = D3D8VBO {buffer, Stride};
```

Comment:
Why is this necessary?

Reply:
It's not. I can remove it, but comparing is a bit faster then COM assignment shenanigans and I've seen d3d9 do the same. Nothing actually depends on it.